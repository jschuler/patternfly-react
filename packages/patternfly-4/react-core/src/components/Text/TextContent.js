import React from 'react';
import styles from '@patternfly/patternfly-next/components/Content/content.css';
import { StyleSheet, css } from '@patternfly/react-styles';
import PropTypes from 'prop-types';

const propTypes = {
  /** Content rendered within the TextContent */
  children: PropTypes.node,
  /** Additional classes added to the TextContent */
  className: PropTypes.string
};

const defaultProps = {
  children: null,
  className: ''
};

const TextContent = ({ children, className, ...props }) => {
  // Get the stylesheet and make it more specific by appending .is-content class to selectors
  // This way even if other components are nested within the TextContent, their text styling will not be affected
  const moreSpecificStyles = styles.raw.replace(/(.pf-c-content\s[a-zA-Z0-9]+)/g, '$1.is-content');
  const updatedStyles = StyleSheet.parse(moreSpecificStyles);

  return (
    <div {...props} className={css(updatedStyles.content, className)}>
      {children}
    </div>
  )
};

TextContent.propTypes = propTypes;
TextContent.defaultProps = defaultProps;

export default TextContent;
