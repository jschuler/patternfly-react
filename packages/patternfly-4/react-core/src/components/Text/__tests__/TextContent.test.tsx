// Auto-generated do not edit

import * as React from 'react';
import { shallow } from 'enzyme';
import { TextContent } from '../TextContent';

describe('TextContent test', () => {
  it('TextContent should match snapshot', () => {
    const view = shallow(
      <TextContent
        children={<>ReactNode</>}
				className={"''"}
      />);
    expect(view).toMatchSnapshot();
  });
});
