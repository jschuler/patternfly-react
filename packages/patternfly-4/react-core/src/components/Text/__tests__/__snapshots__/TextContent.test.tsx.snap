// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextContent test TextContent should match snapshot 1`] = `
<div
  className="pf-c-content ''"
>
  ReactNode
</div>
`;
